extensions/wikia/AdEngine/js/video/player/porvata/porvataPlayerFactory.js

```diff
@@ -103,7 +106,24 @@ define('ext.wikia.adEngine.video.player.porvata.porvataPlayerFactory', [
 				width = newWidth;
 				height = newHeight;
 
-				ima.resize(width, height);
+				if (params.isFullscreen) {
+					ima.resize(ima.viewMode.FULLSCREEN);
+				} else {
+					ima.resize(ima.viewMode.NORMAL, width, height);
+				}
+			},
+			toggleFullscreen: function () {
+				if (params.isFullscreen) {
+					ima.resize(ima.viewMode.NORMAL, width, height);
+					this.container.classList.add(videoFullscreenClassName);
+					doc.documentElement.classList.remove(stopScrollingClassName);
+					params.isFullscreen = false;
+				} else {
+					ima.resize(ima.viewMode.FULLSCREEN);
+					this.container.classList.remove(videoFullscreenClassName);
+					doc.documentElement.classList.add(stopScrollingClassName);
+					params.isFullscreen = true;
+				}
```

Comment:
What do you think about moving lines 120 and 125 outside the `if ... else` block and set
```
params.isFullscreen = !params.isFullscreen;
```

Reply:
ok